Mathlib/Data/Fin/Rev.lean

```diff
@@ -0,0 +1,112 @@
+/-
+Copyright (c) 2017 Robert Y. Lewis. All rights reserved.
```

Comment:
`Fin.rev` came from https://github.com/leanprover-community/mathlib3/pull/12792 or https://github.com/leanprover-community/mathlib3/pull/16357 or https://github.com/leanprover-community/mathlib3/pull/16745; let's give @ericrbg the credit and list Joel and Yury in the authors list?